core/model/__init__.py

```diff
@@ -290,12 +293,25 @@ def dump_query(query):
 DEBUG = False
 
 
-def acquire_advisory_lock(connection, lock_id):
-    connection.execute(f"SELECT pg_advisory_lock({lock_id});")
+@contextmanager
+def pg_advisory_lock(connection: Session, lock_id: int | None):
```

Comment:
Minor: Might want to add a return annotation, so this function counts as fully annotated:
`def pg_advisory_lock(connection: Session, lock_id: int | None) -> None:`